dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -247,15 +247,35 @@ def stage_extras
       return
     end
 
-    @stage = Script.get_from_cache(params[:script_id]).lesson_by_relative_position(params[:stage_position].to_i)
+    user = @user || current_user
+    @stage = Script.get_from_cache(
+      params[:script_id]
+    ).lesson_by_relative_position(
+      params[:stage_position].to_i
+      )
     @script = @stage.script
+    script_bonus_levels_by_stage = @script.get_bonus_script_levels(@stage)
+
+    # get user progress for bonus levels
```

Comment:
I think it'd be good to elaborate on this comment a bit and indicate why we're doing this here.  Also, this loop now also sets the localized display name so may want to move/adjust this comment.